backtrace-library/src/main/java/backtraceio/library/breadcrumbs/BacktraceBroadcastReceiver.java

```diff
@@ -0,0 +1,95 @@
+package backtraceio.library.breadcrumbs;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import backtraceio.library.base.BacktraceBase;
+import backtraceio.library.enums.BacktraceBreadcrumbType;
+import backtraceio.library.logger.BacktraceLogger;
+
+public class BacktraceBroadcastReceiver extends BroadcastReceiver {
+
+    private BacktraceBreadcrumbs backtraceBreadcrumbs;
+
+    private static transient String LOG_TAG = BacktraceBroadcastReceiver.class.getSimpleName();
+
+    public BacktraceBroadcastReceiver(@NonNull BacktraceBreadcrumbs backtraceBreadcrumbs)
+    {
+        this.backtraceBreadcrumbs = backtraceBreadcrumbs;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String action = intent.getAction();
+        if (action == null)
+        {
+            action = "Null action received. This is a bug";
+            BacktraceLogger.e(LOG_TAG, action);
```

Comment:
should we continue when action is null?

Reply:
Good point, we can ignore the null action. Will make this change!